src/Agent.Plugins/PipelineCache/PipelineCacheServer.cs

```diff
@@ -39,6 +39,24 @@ public class PipelineCacheServer
 
             using (clientTelemetry)
             {
+                // Check if the key exists.
+                GetPipelineCacheArtifactOptions getOptions = new GetPipelineCacheArtifactOptions
+                {
+                    Key = key,
+                    Salt = salt,
+                };
+                PipelineCacheActionRecord cacheRecordGet = clientTelemetry.CreateRecord<PipelineCacheActionRecord>((level, uri, type) =>
+                        new PipelineCacheActionRecord(level, uri, type, PipelineArtifactConstants.RestoreCache, context));
+                PipelineCacheArtifact getResult = await pipelineCacheClient.GetPipelineCacheArtifactAsync(getOptions, cancellationToken, cacheRecordGet);
```

Comment:
If we are only checking whether CacheArtifact object exists, do we really need to get the entire object? Or is it because we only have this function currently in the CacheClient to check? :)